### PR TITLE
Fixed problem with repeating instruments / events

### PR DESCRIPTION
**Pull request by jwalsoe**
I experienced problems with repeating events / instruments, which are fixed in this revision. I have not changed config files or version numbers, just some changes in the main php file.

### CODE REVIEW

File: ImageViewer.php [hunk 1]
```diff
@@ -250,12 +285,14 @@ function renderPreview($instrument, $record, $event_id) {
         }
 
         // We need to know the filetype to validate when the file has been previously uploaded...
-        $q = REDCap::getData('json',$record, array_keys($fields), $event_id);
-        $results = json_decode($q, true);
-        $result = $results[0];
+        /*$q = REDCap::getData('json',$record, array_keys($fields), $event_id);
```
Comment (123andy) [inline ImageViewer.php]: @jwalsoe - is it not possible to obtain the doc_id from the standard getData by parsing out the repeat instance value?  Just curious as to why you had to add the sql lookup function.

Comment (jwalsoe) [reply ImageViewer.php]: Maybe it is possible, I just looked at the getData documentation and saw that it had no parameter for repeat instance, and I was in a hurry (one project needed this asap). For me -this time- using the sql lookup was the quickest way to get this done. I have much better tools for working with the database than for working towards the hook/plugin functions. It probably is a better idea to rewrite this stuff to just use the getData output. I might venture to do this once I have better time, provided no one else does it before me :-)

**Comment by jwalsoe** [comment]
Ok, reworked this now to use stuff from getData. Removed SQL lookup. Very small changes to original actually.

**Comment by jwalsoe** [comment]
Closing

File: ImageViewer.php [hunk 2]
```diff
@@ -285,14 +252,14 @@ function renderPreview($instrument, $record, $event_id, $repeat_instance, $proje
         }
 
         // We need to know the filetype to validate when the file has been previously uploaded...
-        /*$q = REDCap::getData('json',$record, array_keys($fields), $event_id);
-        *$results = json_decode($q, true);
-        *$result = $results[0];
-        */
+        $q = REDCap::getData('json',$record, array_keys($fields), $event_id);
+        $results = json_decode($q, true);
+        $result = $results[$repeat_instance - 1];
```
Comment (123andy) [inline ImageViewer.php]: That's interesting.. I haven't played around that much with repeat instances, but worry about a scenario where someone has three repeat instances and then deletes instance '2'.  Now, they go to do a preview when viewing instance 3.  In this case, isn't the index of the results array going to be off?  

I think one would have to actually iterate for a match to handle this scenario... @jwalsoe

Comment (jwalsoe) [reply ImageViewer.php]: You are absolutely on the money with that worry. That is why I closed the pull request. The SQL lookup works fine though. I was testing more around repeat instances and found a bug in RC yesterday, which I rapported in, seems to be a case of ticket #35814 which still has a problem when first instance is deleted. Also has consequences for file upload fields in other instances,
 where file reference gets out of whack (deleting instances does not completely remove it from dash).

**Pull request closed without merging**